### PR TITLE
Add texhyphj to the dotify codebase with a small bugfix.

### PR DESCRIPTION
**Pull request by kalaspuffar**
Hi @bertfrees and @PaulRambags 

We are currently experiencing a problem with production containing Russian texts that crash when we try to create braille. It seems like the hyphenation library used can't load tex files where the words don't have any hyphens. 

If no characters are marked, the new array will be longer than the original, and we get an index out-of-bounds issue.

The fix I've done is to add three lines to `extractExceptionHyphenation` to ensure that we don't map strings without hyphens.

```
        if (!exceptedWord.contains("-")) {
            return tmpHyphenations;
        }
```

Moreover, I've added the library to the dotify code. The rationale is that the original code from David Tolpin is unchanged, but Joel has added most of the code that Dotify requires. Maintaining this in a separate library where the use case for this modified code is limited might not be warranted. 

Other changes from the original code are to follow our style guide. I've added some javadoc, and added brackets and spaces where required.

Best regards
Daniel

### CODE REVIEW

File: src/net/davidashen/text/Utf8TexParser.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+            }
+        }
+
+        return buffer.toString();
+    }
+
+    private int[] extractExceptionHyphenation(String exceptedWord) {
+        int[] tmpHyphenations = new int[exceptedWord.length()];
+
+        if (!exceptedWord.contains("-")) {
```
Comment (PaulRambags) [inline src/net/davidashen/text/Utf8TexParser.java:145]: Please mention that you've added this in a comment

**Review by PaulRambags — CHANGES_REQUESTED** [review]
There are no unit tests at all...
The `src/net/davidashen` library is not really used here. Maybe it could be removed from dotify.library and put into Dotify itself?

**Comment by bertfrees** [comment]
Hum, I'm not so sure about adding the code to the Dotify repo. Not sure I understand the rationale. I don't think Joel made it specific to the braille use case, did he? Yes, he adapted the library specifically for Dotify, but that for me isn't a good enough reason to integrate it. It's still someone else's code originally. Ideally I want as little code as possible (that isn't ours) in the repo, so I'm cautious... The fact that there are no unit tests, as noted by Paul, doesn't help. If it's only to make it more convenient to do bug fixes like this one, which I expect to happen rarely, I vote against it.

If we do add it to Dotify I would separate the initial import from the bug fix. Or, like Paul said, a comment that explains you are the author of those three lines. Basically just some way of telling where exactly which code came from. I guess that also means stating in package-info.java which revision was taken. Finding out where exactly code came from is often a challenge in these cases when something is copied from elsewhere.

**Comment by kalaspuffar** [comment]
This PR will be replaced with https://github.com/mtmse/dotify.library/pull/52 if you agree with that solution.

**Pull request closed without merging**